floris/floris_model.py

```diff
@@ -210,6 +210,22 @@ def _reinitialize(
         if air_density is not None:
             flow_field_dict["air_density"] = air_density
         if heterogeneous_inflow_config is not None:
+
+            # If both heterogeneous_inflow_config contains a z term,
+            # and flow_field_dict["wind_shear"] is
+            # not 0, raise and error the user that there is potential
+            # for double-counting of shear
+            if (
+                "z" in heterogeneous_inflow_config
+                and flow_field_dict["wind_shear"] != 0.0
+            ):
+                if heterogeneous_inflow_config['z'] is not None:
```

Comment:
I don't think there need to be nested `if` statements here? Can we just do:
IF "z" in heterogeneous_inflow_config AND h_i_c["z"] is not None AND 0 shear:
    raise ValueError()
    
?

Reply:
I think if z is not in heterogeneous_inflow_config, then heterogeneous_inflow_config['z'] raises an error

Reply:
I'd have to check, but I think it will check the statements in order---it doesn't execute the second clause of the AND if the first is already false

Reply:
I think I like it a shade better this way, but not so attached if you have a strong preference to one line

Reply:
Just checked, and it will do them in order, for instance:
```
x = {"a": 2}

if "b" in x and x["b"] == 2:
    print("Hello, world")
```
runs as expected (no printout, and no error raised)

Reply:
Done!